src/Orleans/Core/IStatefulGrain.cs

```diff
@@ -0,0 +1,11 @@
+﻿using Orleans.Core;
+
+namespace Orleans
+{
+    internal interface IStatefulGrain
```

Comment:
The new interface makes easy to check if grain is stateful, e.g at https://github.com/dotnet/orleans/pull/1060/files?diff=unified#diff-bcfe9efaf8ce10fd652fca0dd5a4d116R642. I wonder if there's better way of doing this, and implementing https://github.com/dotnet/orleans/pull/1060#issuecomment-162099800 will not remove the need in this interface.